JiscSD/octopus
# OCT-354

The purpose of this PR was to allow a user to edit a reference once it has been added.

### Acceptance Criteria:

- Edits are made in a rich text editor, which opens in a modal view. 
- User must click ‘save’ or ‘cancel’ to close the modal. 
- On save, the text field and URL field are updated. 
- The URL column is not directly editable. 
- User can edit text formatting e.g. underline, bold. Accents and non-european characters should be accepted. 
- User cannot reorder references 
- 
### Tests:

1. Add references to publication
2. Preview references and return to editing
3. Click the 'edit' button to the left of the trash icon
4. Change a particular reference and save

Expected:
Reference should update and save upon saving or pressing preview.

### Screenshots:

![image](https://user-images.githubusercontent.com/69204118/188897516-0373c89d-e6a7-4709-835f-9c7b10442064.png)
